File: remove-duplicates-from-sorted-array_26/remove-duplicates-from-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        k = 0 # This is our writer
        for i in range(1,len(nums)) : # This is our reader
            if nums[k] != nums[i] : 
                k=k+1
                nums[k] = nums[i]
        return k+1
    # L'idee fondamentale est d'avoir un fast et un slow pointer .Le fast pointer permet de parcourir l'array et de trouver les elements uniques alors que le slow pointer permet de overwrite les elements du tableau a gauche et de retourner la taille du tableau sans les duplicates
    # k+1 parce que k est un index et nous on veut la taille du tableau (Toujour taille tableau =index +1)

solution = Solution()
print(solution.removeDuplicates([1,1,2,2,3,4,4,5]))

File: remove-duplicates-from-sorted-array_26/remove-duplicates-from-sorted-array.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public : 
    int removeDuplicates(vector<int>& nums) {
        int k = 0;
        for (int i = 1 ; i< nums.size() ; i++){
                if(nums[k] != nums[i]){
                    k++;
                    nums[k] = nums[i] ;
                }
            }
        return k+1 ;
    }
};

int main() {
    Solution solution;
    vector<int> nums = {1, 1, 2, 2, 3};
    int result = solution.removeDuplicates(nums);
    cout << "Nouveau k : " << result << endl;
    return 0;
}